### PR TITLE
Make tests compile when used as input to RholangCLI

### PR DESCRIPTION
**Pull request by birchmd**
-All these tests should now compile, except for the wildcards (`_`) used in `match` statements, which still needs to be fixed

### CODE REVIEW

File: rholang/src/main/bnfc/rholang_mercury.cf [hunk 1]
```diff
@@ -51,13 +51,13 @@ PIfElse.      Proc1 ::= "if" "(" Proc ")" Proc2 "else" Proc1 ;
 PNew.         Proc1 ::= "new" [NameDecl] "in" Proc1 ;
 PPar.         Proc  ::= Proc "|" Proc1 ;
 
-separator nonempty Proc "," ;
+separator Proc "," ;
```
Comment (KentShikama) [inline rholang/src/main/bnfc/rholang_mercury.cf]: I don't see any usage below of having an empty terminator for a Proc. Are we intentionally adding this as a feature of the language?

File: rholang/src/main/bnfc/rholang_mercury.cf [hunk 2]
```diff
@@ -51,13 +51,13 @@ PIfElse.      Proc1 ::= "if" "(" Proc ")" Proc2 "else" Proc1 ;
...
 
 -- Names
 NameWildcard. Name ::= "_" ;
 NameVar.      Name ::= Var ;
 NameQuote.    Name ::= "@" Proc12 ;
-separator nonempty Name "," ;
+separator Name "," ;
```
Comment (KentShikama) [inline rholang/src/main/bnfc/rholang_mercury.cf]: Same as above

File: rholang/src/main/bnfc/rholang_mercury.cf (hunk 1 above)
Comment (birchmd) [reply rholang/src/main/bnfc/rholang_mercury.cf]: It's to make the empty list `[]` compile. Removing nonempty from proc and name separators was @guardbotmk3 's suggestion

File: rholang/src/main/bnfc/rholang_mercury.cf (hunk 2 above)
Comment (guardbotmk3) [reply rholang/src/main/bnfc/rholang_mercury.cf]: This enables empty lists, and empty sends/receives. I told him to do it.

**Pull request merged**